src/bus/mod.rs

```diff
@@ -64,8 +64,14 @@ impl Bus {
             Transport::Unix(unix) => {
                 // Resolve address specification into address that clients can use.
                 let addr = Self::unix_addr(unix)?;
-
-                (Self::unix_stream(addr).await?, AuthMechanism::External)
+                address = Address::try_from(
+                    format!("unix:path={}", addr.as_pathname().unwrap().display()).as_str(),
+                )?;
```

Comment:
So sorry but I did point out this before here: You don't (at least should not) need any parsing here.

Reply:
See [this comment from last month](https://github.com/dbus2/busd/pull/182#discussion_r1885747383).